CondFormats/SiPixelTransient/src/SiPixelUtils.cc

```diff
@@ -71,17 +71,95 @@ namespace SiPixelUtils {
     }
 
     //--- Finally, compute the position in this projection
-    float qdiff = q_l - q_f;
-    float qsum = q_l + q_f;
+    float q_diff = q_l - q_f;
+    float q_sum = q_l + q_f;
 
     //--- Temporary fix for clusters with both first and last pixel with charge = 0
-    if (qsum == 0)
-      qsum = 1.0f;
+    if (q_sum == 0)
+      q_sum = 1.0f;
 
-    //float hit_pos = geom_center + 0.5f*(qdiff/qsum) * w_eff + half_lorentz_shift;
-    float hit_pos = geom_center + 0.5f * (qdiff / qsum) * w_eff;
+    //float hit_pos = geom_center + 0.5f*(q_diff/q_sum) * w_eff + half_lorentz_shift;
+    float hit_pos = geom_center + 0.5f * (q_diff / q_sum) * w_eff;
 
     return hit_pos;
   }
 
+  float generic_position_formula_y_bricked(
+      int size,                    //!< Size of this projection.
+      int q_f,                     //!< Charge in the first pixel.
+      int q_l,                     //!< Charge in the last pixel.
+      int q_f_b,                   //!< Charge in pixels that are "dented" compared to the lowest pixel of the cluster.
+      int q_l_b,                   //!< Charge in pixels that are "dented" compared to the highest pixel of the cluster.
+      float upper_edge_first_pix,  //!< As the name says.
+      float lower_edge_last_pix,   //!< As the name says.
+      float lorentz_shift,         //!< L-shift at half thickness
+      float theThickness,          //detector thickness
+      float cot_angle,             //!< cot of alpha_ or beta_
+      float pitch,                 //!< thePitchX or thePitchY
+      bool first_is_big,           //!< true if the first is big
+      bool last_is_big,            //!< true if the last is big
+      float eff_charge_cut_low,    //!< Use edge if > w_eff  &&&
+      float eff_charge_cut_high,   //!< Use edge if < w_eff  &&&
+      float size_cut               //!< Use edge when size == cuts
+  ) {
+    //cout<<" in PixelCPEGeneric:generic_position_formula - "<<endl; //dk
+
+    float geom_center = 0.5f * (upper_edge_first_pix + lower_edge_last_pix);
+
+    //--- The case of only one pixel in this projection is separate.  Note that
+    //--- here first_pix == last_pix, so the average of the two is still the
+    //--- center of the pixel.
+
+    //Make use of the bricked geometry, comment this out
+    //if (size == 1) {
+    //return geom_center;
+    //}
```

Comment:
I thought this PR is making use of the bricked geometry, so should this be uncommented?

Reply:
Hi! This should not be uncommented as in the bricked geometry, the size defined as the difference in columns does not mean that the two pixels are aligned. So in this case too, there needs to be an analysis.